src/OrchardCore.Modules/OrchardCore.Sitemaps/Controllers/SitemapController.cs

```diff
@@ -15,7 +15,7 @@ namespace OrchardCore.Sitemaps.Controllers
 {
     public class SitemapController : Controller
     {
-        private static readonly ConcurrentDictionary<string, Lazy<Task<Stream>>> Workers = new ConcurrentDictionary<string, Lazy<Task<Stream>>>();
+        private static readonly ConcurrentDictionary<string, Lazy<Task<Stream>>> Workers = new ConcurrentDictionary<string, Lazy<Task<Stream>>>(StringComparer.OrdinalIgnoreCase);
```

Comment:
Why using `StringComparer.OrdinalIgnoreCase` as the keys are unique identifiers.

Reply:
I think you did a mistake with `_sitemapPaths` and `_sitemapIds`

E.g in `_sitemapPaths` this is the keys that are paths

I would want to fix it in my distributedCache branch but maybe a breaking change

But this is just a convention to tie the name with the keys, not the values as in AutotouteEntries.

So at least, as it is right now, in my branch this is `SitemapPaths` that would need to use `OrdinalIgnoreCase`, not `SitemapIds`. Here you may need to add `OrdinalIgnoreCase` to `_sitemapPaths`.

Reply:
Okay, tested in my distributed branch, good to not use here `OrdinalIgnoreCase` as the keys are unique ids

And good to use `OrdinalIgnoreCase` for the `SitemapPaths` dictionary

Note: First it was not working on my branch because `MessagePack` doesn't preserve the `OrdinalIgnoreCase` comparison when deserializing.